plugins/modules/dcnm_service_policy.py

```diff
@@ -0,0 +1,1962 @@
+        rp_name: it-fw-rp1
+
+      - node_name: it-sn-2
+        rp_name: it-fw-rp2
+
+4. Without config
+
+   Deletes all service policies on the given fabric and attached fabirc
+
+- name: Delete service policies without config
+  cisco.dcnm.dcnm_service_policy:
+    fabric: test_fabirc
+    attached_fabric: external
+    state: deleted
+
+OVERRIDE SERVICE POLICIES
+=========================
+
+- name: Override and delete all service policies
+  cisco.dcnm.dcnm_service_policy:
+    fabric: test_fabric
+    attached_fabric: external
+    state: overridden
+
+- name: Override all existing service policies with a new one
+  cisco.dcnm.dcnm_service_policy:
+    fabric: test_fabric
+    attached_fabric: external
+    attach: true
+    deploy: true
+    state: overridden
+    config:
+      - name: service_policy_1
+        node_name: it-sn-1
+        rp_name: it-fw-rp1
+        src_vrf: vrf_11
+        dest_vrf: vrf_11
+        src_network: net_11
+        dest_network: net_12
+        next_hop: 192.161.1.100
+        rev_next_hop: 192.161.2.100
+        policy:
+          proto: icmp
+          src_port: 555
+          dest_port: 22
+          action: permit
+          next_hop_option:  none
+          acl_name: fwd_acl_555
+          rev_acl_name: rev_acl_555
+          route_map_num: 555
+          rev_route_map_num: 556
+
+REPLACE SERVICE POLICIES
+========================
+
+- name: Replace some of the objects in already created service policy
+  cisco.dcnm.dcnm_service_policy:
+    fabric: test_fabric
+    attached_fabric: external
+    attach: true
+    deploy: true
+    state: replaced
+    config:
+      - name: service_policy_1
+        node_name: it-sn-1
+        rp_name: it-fw-rp1
+        src_vrf: vrf_11
+        dest_vrf: vrf_11
+        src_network: net_11
+        dest_network: net_12
+        next_hop: 192.161.1.100
+        rev_next_hop: 192.161.2.100
+        policy:
+          proto: udp
+          src_port: 501
+          dest_port: 502
+          action: deny
+          next_hop_option: drop_on_fail
+
+QUERY SERVICE POLICIES
+======================
+
+- name: Query service policies based on service node and policy name
+  cisco.dcnm.dcnm_service_policy:
+    fabric: test_fabric
+    attached_fabric: external
+    state: query
+    config:
+      - name: service_policy_1
+        node_name: it-sn-1
+
+- name: Query service policies based on service node
+  cisco.dcnm.dcnm_service_policy:
+  fabric: test_fabric
+  attached_fabric: external
+  state: query
+  config:
+    - node_name: it-sn-1
+
+"""
+
+import time
+import json
+import copy
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.dcnm.plugins.module_utils.network.dcnm.dcnm import (
+    dcnm_send,
+    validate_list_of_dicts,
+    dcnm_reset_connection,
+)
+
+from datetime import datetime
+
+
+# Service Policy Class object which includes all the required methods and data to configure and maintain Roue peering objects
```

Comment:
`/Roue/Route/`

But should it read Service Policy Objects instead?

Reply:
I have replaced Route Peerings with Service Policies in a few places. As part of that this too would have been replaced.